app/src/main/java/com/alphawallet/app/repository/TokensRealmSource.java

```diff
@@ -595,23 +595,36 @@ private boolean traitsDifferent(List<Trait> traits, List<Trait> traits1)
         return assets;
     }
 
```

Comment:
Maybe create a new method to fetch all the TCMs (fetchAllTokenMetas); expanding the existing one is heading for trouble.